conn_test.go

```diff
@@ -314,9 +345,8 @@ func TestQueryRetry(t *testing.T) {
 		t.Fatalf("expected requests %v to match query attempts %v", requests, attempts)
 	}
 
-	// the query will only be attempted once, but is being retried
-	if requests != int64(rt.NumRetries) {
-		t.Fatalf("failed to retry the query %v time(s). Query executed %v times", rt.NumRetries, requests-1)
+	if requests > 8 {
```

Comment:
this will never reach 9 retries because the `QueryTimeout` of `200ms` will stop the query before.

Reply:
i have added a comment about it

Reply:
yep, way more clear. Thanks!

Reply:
Yeah, this is a bit of a mess. But we couldn't figure out how to make it any less messy because there's no way in the test to pass the attempt timeout to the framer (tbh, I really hate that test code). Also, if we do keep it we should fail the test if there aren't any retries because right now it only fails if there's too many. Ideally, the reasoning should be improved by asserting those retries at the framer level but as I said, that's currently not possible. :/

Reply:
I really really like that suggestion and implemented it right away! This feels so much better. I didn't add the `AttemptTimeout` method to the RetryPolicy interface in order to not break existing code. Take a look?

Reply:
I also added a guard assertion to make sure the query is retried.

Reply:
It looks way way better now. I even like how you implemented the RetryPolicyWithTimeout separately. It's a neat trick, will def bookmark it for myself.

Everything is so much cleaner now and totally makes sense.

Reply:
just let's make sure it builds cleanly. I saw the error in the build, totally not your fault :-)

Reply:
I'll give it another shot now.